src/ViewRendererBundle/objects/helper/TPkgViewRendererLessCompiler.php

```diff
@@ -191,25 +191,42 @@ public function getGeneratedCssForPortal($portal, $minifyCss = null)
                 $filesForLessParsing[PATH_WEB.$lessFile] = '/';
             }
 
-            \Less_Cache::SetCacheDir($cachedLessDir);
-            try {
-                $cssFile = \Less_Cache::Get($filesForLessParsing, $options);
-            } catch (Exception $exc) {
-                if (false !== strpos($exc->getMessage(), 'stat failed')) {
-                    // Consider this as a 'File removed! Halp!' and clear the cache and try again
-                    array_map('unlink', glob($cachedLessDir.'/*'));
+            $css = '';
 
+            if (_DEVELOPMENT_MODE) {
+                \Less_Cache::SetCacheDir($cachedLessDir);
+                try {
                     $cssFile = \Less_Cache::Get($filesForLessParsing, $options);
-                } else {
+                } catch (Exception $exc) {
+                    if (false !== strpos($exc->getMessage(), 'stat failed')) {
+                        // Consider this as a 'File removed! Halp!' and clear the cache and try again
+                        array_map('unlink', glob($cachedLessDir.'/*'));
+
+                        $cssFile = \Less_Cache::Get($filesForLessParsing, $options);
+                    } else {
+                        throw new ViewRenderException('Exception during less compile', 0, $exc);
+                    }
+                }
+
+                $absoluteCssFilepath = $cachedLessDir.'/'.$cssFile;
+
+                $css = file_get_contents($absoluteCssFilepath);
+            } else {
+                try {
+                    $parser = new \Less_Parser($options);
+                    foreach ($filesForLessParsing as $file => $root) {
+                        $parser->parseFile($file, $root);
+                    }
+
+                    $css = $parser->getCss();
```

Comment:
Just to be sure: we only ever get here when the generated css does not exist, correct?

Reply:
If `chameleon_system_core.cache.cache_less_files` is true.